src/consensus/aft/raft.h

```diff
@@ -1174,12 +1199,14 @@ namespace aft
       }
 
       // Then check if those append entries extend past our retirement
-      if (
-        retirement_committable_idx.has_value() &&
-        r.idx > retirement_committable_idx && !is_retiring())
+      if (is_retired() && retirement_phase >= kv::RetirementPhase::Completed)
```

Comment:
I believe this condition is too strong, but I got into trouble with the 1tx scheme without this. Shouldn't it be `>= ...::Signed`?

Reply:
No, we still want to ack the signature over our retirement (because we may hold up commit otherwise), just nothing (strictly) further.